### PR TITLE
ROX-12319: implement smoke test step with groovy test filter

### PR DESCRIPTION
**Pull request by jschnath**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: scripts/ci/jobs/gke_version_compatibility_tests.py [hunk 1]
```diff
@@ -3,5 +3,19 @@
 """
 Run version compatibility tests
 """
+import os
+from compatibility_test import compatibility_test
```
Comment (fredrb) [inline scripts/ci/jobs/gke_version_compatibility_tests.py:7]:
```suggestion
from compatibility_test import make_compatibility_test_runner
```

**Comment by fredrb** [comment]
Could we add some indication that a test failure happens with a particular sensor version? Right now I see a failure but there is not easy way to figure out in which version configuration it ran.

File: qa-tests-backend/src/test/groovy/AutocompleteTest.groovy [hunk 2]
```diff
@@ -14,7 +15,7 @@ class AutocompleteTest extends BaseSpecification {
 
     private static final String GROUP_AUTOCOMPLETE = isPostgresRun() ? "GROUP" : "group"
 
-    @Category([BAT])
+    @Category([BAT, COMPATIBILITY])
```
Comment (fredrb) [inline qa-tests-backend/src/test/groovy/AutocompleteTest.groovy:18]: Remove compatibility

File: qa-tests-backend/src/test/groovy/AutocompleteTest.groovy [hunk 3]
```diff
@@ -39,7 +40,7 @@ class AutocompleteTest extends BaseSpecification {
     }
 
     @Unroll
-    @Category([BAT])
+    @Category([BAT, COMPATIBILITY])
```
Comment (fredrb) [inline qa-tests-backend/src/test/groovy/AutocompleteTest.groovy:43]: Remove compatibility

**Comment by fredrb** [comment]
@jschnath seems like tests passed but post test failed 🤔 

```
ERROR: post test failed [version compatibility tests]
```

**Comment by jschnath** [comment]
And that's 5 successful prow runs in a row after the rebase to master. I think this is ready for merging @fredrb

**Review by fredrb — APPROVED** [review]

**Pull request merged**